internal/repository/rsl.go

```diff
@@ -139,18 +142,197 @@ func (r *Repository) RecordRSLAnnotation(rslEntryIDs []string, skip bool, messag
 	return rsl.NewAnnotationEntry(rslEntryHashes, skip, message).Commit(r.r, signCommit)
 }
 
+// ReconcileLocalRSLWithRemote checks the local RSL against the specified remote
+// and reconciles the local RSL if needed. If the local RSL doesn't exist or is
+// strictly behind the remote RSL, then the local RSL is updated to match the
+// remote RSL. If the local RSL is ahead of the remote RSL, nothing is updated.
+// Finally, if the local and remote RSLs have diverged, then the local only RSL
+// entries are reapplied over the latest entries in the remote.
```

Comment:
Note: for now, this doesn't do anything special if a new entry in the remote is for the same ref as one of the local only refs. We need to do a little more work to assess when we can encounter these cases. My hunch is this is likely to be for gittuf specific refs, because otherwise, there shouldn't be an unpushed RSL entry in the local RSL. To be safe, I can update this to fail if the set of refs in the remote only entries overlap with the set of refs in the local only entries.

Reply:
Just pushed  a change to this effect.

Reply:
Actually, even if this is just happening on gittuf refs, it's probably better for the human to intervene. If it's because two people tried to change the policy at the same time, we don't want one to wipe out the other's change by overwriting the tree in the RSL.

Reply:
At the same time, for high traffic cases where people may be pushing to different branches, this is great.